HIP/hip-1064.md

```diff
@@ -0,0 +1,163 @@
+
+```
+/**
+ * A consensus snapshot.<br/>
+ * This is a snapshot of the consensus state for a particular round.
+ *
+ * This message SHALL record consensus data necessary for restart
+ * and reconnect.
+ */
+message ConsensusSnapshot {
+    ....
+    /**
+     * A list of SHA-384 hash values.<br/>
+     * The hashes of all judges for this round.
+     * <p>
+     * This list SHALL be ordered by creator ID.<br/>
+     * This list MUST be deterministically ordered.
+     */
+    repeated bytes judge_hashes = 2 [deprecated=true];
+    ....
+    /*
+     * A list of judges.
+     * This list SHALL be ordered by creator ID.<br/>
+     * This list MUST be deterministically ordered.
+     */
+    repeated Judge judges = 6;
+}
+
+/**
+ * A judge information
+ */
+message Judge {
+  /**
+   * The creator node ID who created this judge.
+   */
+    uint64 creator_id = 1;
+    /**
+     * SHA-384 hash value of this judge
+     */
+    bytes judge_hash = 2;
+}
+```
+
+- To store the number of rounds in a staking period and the number of rounds in which each node created judges, a new singleton state `NodeRewards`will be added to `TokenService` . This singleton state will be updated at the end of every round.
+
+```
+syntax = "proto3";
+
+package proto;
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.state.token">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * A node rewards message. This is used to find number of active nodes in a staking period based on number of judges
+ * each node created in that period.
+ */
+message NodeRewards {
+    /**
+     * Number of rounds in a staking period.
+     */
+    uint64 num_rounds_in_staking_period = 1;
+    /**
+     * Number of judges created by each node in a staking period.
+     */
+    repeated NodeActivity node_activities = 2;
+}
+/**
+  * A node activity containing number of judges created in a staking period.
+ */
+message NodeActivity {
+    /**
+     * Node ID of the node.
+     */
+    uint64 node_id = 1;
+    /**
+     * Number of judges created by the node.
+     */
+    uint64 num_judge_rounds = 2;
+}
+```
+
+## Backward Compatibility
+
+Existing nodes will not require any changes to their software or configuration to be eligible for rewards, as long as they meet the active criteria.
+
+## Security Implications
+
+No known security concerns.
+
+## How to Teach This
+
+This HIP incentivizes node operators for keeping the node active by guaranteeing a minimum reward in the periods that they accomplish this.
+
+## Rejected Ideas
+
+- Considered using number of submitted transactions to node to treat the node as active.
```

Comment:
```suggestion
- We considered using the number of transactions submitted by a node to assess node activity.
   - This was rejected in favor of counting rounds with judges because the latter
     approach incorporates consensus network assessment of node "honesty" and
     is more "fair".
```